packages/next/src/server/lib/patch-fetch.ts

```diff
@@ -298,31 +298,54 @@ export function createPatchedFetcher(
           typeof currentFetchCacheConfig === 'string' &&
           typeof currentFetchRevalidate !== 'undefined'
         ) {
-          // when providing fetch with a Request input, it'll automatically set a cache value of 'default'
-          // we only want to warn if the user is explicitly setting a cache value
-          if (!(isRequestInput && currentFetchCacheConfig === 'default')) {
+          // If the revalidate value conflicts with the cache value, we should warn the user and unset the conflicting values.
+          const isConflictingRevalidate =
+            // revalidate: 0 and cache: force-cache
+            (currentFetchCacheConfig === 'force-cache' &&
+              currentFetchRevalidate === 0) ||
+            // revalidate: >0 and cache: no-store
+            (currentFetchCacheConfig === 'no-store' &&
+              currentFetchRevalidate > 0)
```

Comment:
Does this also need to handle the `false` case or is it normalized by this point?

Reply:
Forgot this was an option, fixed